plugins/gssapi.c

```diff
@@ -811,6 +829,229 @@ gssapi_server_mech_new(void *glob_context,
     return SASL_OK;
 }
 
+#ifdef GSS_USE_CC
+static char* strsubst(context_t* text, const char* haystack, const char* needle, const char* subst) {
+  const size_t haystack_len = strlen(haystack);
+  const size_t needle_len = strlen(needle);
+  const size_t subst_len = strlen(subst);
+  const size_t dest_len = haystack_len - needle_len + subst_len;
+  const char* needle_ptr = strstr(haystack, needle);
+  const size_t pre_needle_len = needle_ptr - haystack;
+  char* d;
+
+  if ((NULL == haystack) || (NULL == needle) || (NULL == subst)) {
+    text->utils->seterror(text->utils->conn, SASL_LOG_ERR,
+                          "GSSAPI error: Bad ccache format");
+    return NULL;
+  }
+
+  if (NULL == (d = malloc(dest_len + 1))) {
+    text->utils->seterror(text->utils->conn, SASL_LOG_ERR,
+                          "GSSAPI error: Out of memory processing ccache");
+    return NULL;
+  }
+
+  strncpy(d, haystack, pre_needle_len);
+  d[pre_needle_len] = 0;
+  strcat(d, subst);
+  strcat(d, &haystack[pre_needle_len + needle_len]);
+
+  return d;
+}
+
+static char* create_krb5_ccache(context_t* text, const char* username, const char* ccache) {
```

Comment:
You are not creating the ccache here, simply using a tempate to provide a ccache name, suggest changing name to reflect that and avoid confusion.
Something like resolve_ccache_name()